File: app/src/main/java/com/example/map/domain/repository/FavoriteDocumentRepository.java
package com.example.map.domain.repository;

import com.example.map.data.local.model.DocumentEntity;

import java.util.List;

import io.reactivex.rxjava3.core.Completable;
import io.reactivex.rxjava3.core.Flowable;

public interface FavoriteDocumentRepository {
    Flowable<List<DocumentEntity>> getAll();
    Flowable<DocumentEntity> get(String id);
    Completable insert(DocumentEntity documentEntity);
    Completable delete(DocumentEntity documentEntity);
}

File: app/src/main/java/com/example/map/domain/repository/FavoriteDocumentRepository.kt
package com.example.map.domain.repository

import com.example.map.data.local.model.DocumentEntity
import kotlinx.coroutines.flow.Flow

interface FavoriteDocumentRepository {
    fun getAll(): Flow<List<DocumentEntity>>
    fun get(id: String): Flow<DocumentEntity>
    suspend fun insert(documentEntity: DocumentEntity)
    suspend fun delete(documentEntity: DocumentEntity)
}
